src/identity_committer.rs

```diff
@@ -135,62 +138,98 @@ impl IdentityCommitter {
         // We could use the second element of the tuple as `mut shutdown_receiver`,
         // but for symmetry's sake we create it for every task with `.subscribe()`
         let (shutdown_sender, _) = broadcast::channel(1);
-        let (wake_up_sender, mut wake_up_receiver) = mpsc::channel(1);
+        let (wake_up_sender, wake_up_receiver) = mpsc::channel(1);
         let (pending_identities_sender, pending_identities_receiver) = mpsc::channel(1);
 
-        let process_identities_handle = {
-            let mut shutdown_receiver = shutdown_sender.subscribe();
+        // We need to maintain mutable access to these receivers from multiple
+        // invocations of this task
+        let wake_up_receiver = Arc::new(Mutex::new(wake_up_receiver));
+        let pending_identities_receiver = Arc::new(Mutex::new(pending_identities_receiver));
 
+        let process_identities_handle = {
             let database = self.database.clone();
             let identity_manager = self.identity_manager.clone();
             let batch_tree = self.tree_state.get_batching_tree();
             let timeout = self.batch_insert_timeout_secs;
+            let shutdown_sender = shutdown_sender.clone();
 
-            spawn_or_abort(async move {
-                select! {
-                    result = Self::process_identities(
-                        &database,
-                        &identity_manager,
-                        &batch_tree,
-                        &mut wake_up_receiver,
-                        &pending_identities_sender,
-                        timeout
-                    ) => {
-                        result?;
-                    }
-                    _ = shutdown_receiver.recv() => {
-                        info!("Woke up by shutdown signal, exiting.");
-                        return Ok(());
+            crate::utils::spawn_monitored_with_backoff(
+                move || {
+                    let wake_up_receiver = wake_up_receiver.clone();
```

Comment:
I'm not happy with the second level of clones to move the data into the context - and there definitely is a refactor needed here but I want to move on to the db unification + actually setting up the Task Failed monitoring